nixos/modules/config/fonts/fonts.nix

```diff
@@ -2,6 +2,52 @@
 
 with lib;
 
+let
+  # A scalable variant of the X11 "core" cursor
+  #
+  # If not running a fancy desktop environment, the cursor is likely set to
+  # the default `cursor.pcf` bitmap font. This is 17px wide, so it's very
+  # small and almost invisible on 4K displays.
+  fontcursormisc_hidpi = pkgs.xorg.fontcursormisc.overrideAttrs (old:
+    let
+      # The scaling constant is 230/96: the scalable `left_ptr` glyph at
+      # about 23 points is rendered as 17px, on a 96dpi display.
+      # Note: the XLFD font size is in decipoints.
+      size = 2.39583 * config.services.xserver.dpi;
```

Comment:
finally a use for nix floats :)

Reply:
Yeah, unfortunately (stable?) Nix is lacking a rounding function, so I have to do this ugly string conversion.

Reply:
Ok. I am not really using x11 anymore but maybe @roberth can have another look?

Reply:
Same. Also, gnome applies its own cursor config anyway.
The Nix code looks ok and I think we can trust @rnhmjoj's testing.